src/overlays/actors/ovl_En_Osn/z_en_osn.c

```diff
@@ -74,59 +112,872 @@ static DamageTable D_80AD2550 = {
+    this->collider.dim.pos.x = this->actor.world.pos.x;
+    this->collider.dim.pos.y = this->actor.world.pos.y;
+    this->collider.dim.pos.z = this->actor.world.pos.z;
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
+
+s32 func_80AD08B0(GlobalContext* globalCtx) {
+    switch (Player_GetMask(globalCtx)) {
+        case PLAYER_MASK_GREAT_FAIRY:
+            return 0x1FD6;
+        case PLAYER_MASK_GIBDO:
+            return 0x1FD8;
+        case PLAYER_MASK_TRUTH:
+            return 0x1FDA;
+        case PLAYER_MASK_GIANT:
+            return 0x1FDC;
+        case PLAYER_MASK_KAFEIS_MASK:
+            return 0x1FDE;
+        case PLAYER_MASK_DON_GERO:
+            return 0x1FE0;
+        case PLAYER_MASK_BLAST:
+            return 0x1FE2;
+        case PLAYER_MASK_COUPLE:
+            return 0x1FE4;
+        case PLAYER_MASK_SCENTS:
+            return 0x1FE6;
+        case PLAYER_MASK_KAMARO:
+            return 0x1FE8;
+        case PLAYER_MASK_STONE:
+            return 0x1FEA;
+        case PLAYER_MASK_POSTMAN:
+            return 0x1FEC;
+        case PLAYER_MASK_BUNNY:
+            return 0x1FEE;
+        case PLAYER_MASK_CAPTAIN:
+            return 0x1FF0;
+        case PLAYER_MASK_BREMEN:
+            return 0x1FF2;
+        case PLAYER_MASK_CIRCUS_LEADER:
+            return 0x1FF4;
+        case PLAYER_MASK_KEATON:
+            return 0x1FF6;
+        case PLAYER_MASK_GARO:
+            return 0x1FF8;
+        case PLAYER_MASK_ALL_NIGHT:
+            return 0x1FFA;
+        case PLAYER_MASK_ROMANI:
+            return 0x1FFC;
+        default:
+            return 0;
+    }
+}
+
+void func_80AD0998(EnOsn* this) {
+    s16 sp1E;
+    s16 new_var;
+
+    sp1E = this->anime.curFrame;
+    new_var = Animation_GetLastFrame(sAnimations[this->unk_1EC].animation);
+    if (this->unk_1EC == 0x12 && sp1E == new_var) {
+        this->unk_1EC = 0x13;
+        Actor_ChangeAnimation(&this->anime, sAnimations, 0x13);
+    }
+}
+
+void func_80AD0A24(EnOsn* this) {
+    s16 sp1E;
+    s16 new_var;
+
+    sp1E = this->anime.curFrame;
+    new_var = Animation_GetLastFrame(sAnimations[this->unk_1EC].animation);
+    if (this->unk_1EC == 0x15 && sp1E == new_var) {
+        this->unk_1EC = 0x16;
+        Actor_ChangeAnimation(&this->anime, sAnimations, 0x16);
+    }
+}
+
+void func_80AD0AB0(EnOsn* this) {
+    s16 sp1E;
+    s16 new_var;
+
+    sp1E = this->anime.curFrame;
+    new_var = Animation_GetLastFrame(sAnimations[this->unk_1EC].animation);
+    if (sp1E == new_var) {
+        this->unk_1FC -= 8;
+        if (this->unk_1FC < 8) {
+            this->unk_1FC = 0;
+            Actor_MarkForDeath(&this->actor);
+        }
+    }
+}
+
+s32 func_80AD0B38(EnOsn* this, GlobalContext* globalCtx) {
+    switch (Player_GetMask(globalCtx)) {
+        case 11:
+            if ((this->unk_1F8 & 1) == 0) {
```

Comment:
For bit checks like this we prefer to treat them as boolean instead of checking against zero
```suggestion
            if (!(this->unk_1F8 & 1)) {
```